### PR TITLE
[observability] Alert for high CPU usage.

### PR DESCRIPTION
**Pull request by ArthurSens**
## Description

Adds an alert for high load average. Calculation is done by normalizing 1 min load average over the number of CPU cores that a node has.

## Related Issue(s)

Fixes https://github.com/gitpod-io/runbooks/issues/289

## How to test

Access [Grafana](https://grafana-as-node-cpu-alert.preview.gitpod-dev.com/d/gitpod-overview/gitpod-overview?orgId=1&refresh=30s) to check the normalized load average on the `Gitpod / Overview` dashboard.

Alert can be seen at [Prometheus](https://prometheus-as-node-cpu-alert.preview.gitpod-dev.com/alerts)

## Release Notes

```release-note
NONE
```

## Documentation

- [X] /werft with-observability

### CODE REVIEW

**Comment by ArthurSens** [comment]
/cc @meysholdt 

As predicted, this alert is super noisy. Would have been triggered a little bit less than 20 times in the past hour:
image

**Comment by meysholdt** [comment]
Can we [alert on Load](https://github.com/gitpod-io/observability/issues/289#issuecomment-940979695) instead of CPU? That's a more representative signal for a machine to be "maxed out" and it allows us to set the threshold more find-grained. 

Example: The difference between a "just maxed out" and an "extremely maxed out" machine (assuming one cpu-core) is:
* CPU `99%` vs. `100%`
* Load `0.99` vs. e.g. `20`

On the "load" scale we have far more spectrum.

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/6241#issuecomment-957237048" title="LGTM">JanKoehnlein</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/6241#issuecomment-951058989" title="Approved">meysholdt</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **csweichel** after the PR has been reviewed.

Associated issue: *#289*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- ~~[operations/observability/mixins/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/operations/observability/mixins/OWNERS)~~ [JanKoehnlein,meysholdt]
- **[operations/observability/mixins/workspace/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/operations/observability/mixins/workspace/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by meysholdt** [comment]
> for some reason the observability side of things is not deployed.

Testing this should work again once https://github.com/gitpod-io/gitpod/pull/6458 is merged.

**Comment by roboquat** [comment]
New changes are detected. LGTM label has been removed.

**Comment by ArthurSens** [comment]
Looks like the changes made in  #6458 and #6537 didn't play nicely together. I'll need to make some changes to make monitoring-satellite work on previews again

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/6241#issuecomment-957237048" title="LGTM">JanKoehnlein</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/6241#issuecomment-951058989" title="Approved">meysholdt</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **csweichel** after the PR has been reviewed.

Associated issue: *#289*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- ~~[.werft/observability/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/.werft/observability/OWNERS)~~ [meysholdt]
- ~~[operations/observability/mixins/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/operations/observability/mixins/OWNERS)~~ [JanKoehnlein,meysholdt]
- **[operations/observability/mixins/workspace/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/operations/observability/mixins/workspace/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by ArthurSens** [comment]
Alright, after tons of unrelated bumps, it should be ready for review now :)

**Comment by ArthurSens** [comment]
@csweichel do you think you could take a look at this any time soon?

**Comment by csweichel** [comment]
Like the idea - it's also a nice and understandable signal. We used to have a CPU load view in the overview dashboard, but it
1. suffered from performance problems (afaik solved due to Grafana fixes),
2. was not very helpful because with 20-25 nodes per cluster it turns into an unintelligible mush - we need to find a representation that makes sense on this view (per node we already have dashboards). Maybe showing the `median` and/or `max` per cluster or something like that.

**Comment by ArthurSens** [comment]
@csweichel I've added `topk 5` to the query :)

/werft run with-observability

**Comment by ArthurSens** [comment]
Had some fmt issues 🤦‍♂️ 

/werft run with-observability

**Comment by ArthurSens** [comment]
Another bump 😅, upstream changed the default branch and broken our dependency file. Already fixed at https://github.com/prometheus-operator/kube-prometheus/pull/1491

/werft run with-observability

**Comment by ArthurSens** [comment]
New bump, prometheus-operator getting OOM killed 😪

**Comment by ArthurSens** [comment]
Alright, 5th unrelated blocker solved. @csweichel do you mind taking a look again? :)

File: operations/observability/mixins/cross-teams/dashboards/gitpod-overview.libsonnet [hunk 1]
```diff
@@ -62,54 +62,26 @@ local runningWorkspacesGraph =
...
-        and # We are correlating workspaces with Nodes with high normalized load average
-
-        (
-          sum(
-            instance:node_load1_per_cpu:ratio{cluster=~"$cluster"}
-          ) by (node, %(clusterLabel)s) # Sum to remove all labels except node and %(clusterLabel)s, labels must match with the first part of the query
-        ) > 1
-      ) by (%(clusterLabel)s)
-    ||| % _config, legendFormat='# of workspaces'
+      topk(5, sum(nodepool:node_load1:normalized{%(clusterLabel)s=~"$cluster", nodepool=~".*workspace.*"}) by (node))
```
Comment (meysholdt) [inline operations/observability/mixins/cross-teams/dashboards/gitpod-overview.libsonnet:83]: Loks like Victoriametrics [startet supporting topk](https://docs.victoriametrics.com/MetricsQL.html#topk). This is awesome.

Comment (ArthurSens) [reply operations/observability/mixins/cross-teams/dashboards/gitpod-overview.libsonnet:83]: topk has always worked, the [@ modifier ](https://prometheus.io/docs/prometheus/latest/querying/basics/#modifier)that doesn't :)

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: b88d58574cf636885d1ee6661bd08958a2952dcb</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/6241#issuecomment-968611658" title="LGTM">csweichel</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/6241#issuecomment-957237048" title="LGTM">JanKoehnlein</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/6241#issuecomment-951058989" title="Approved">meysholdt</a>*

Associated issue: *#289*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[.werft/observability/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/.werft/observability/OWNERS)~~ [meysholdt]
- ~~[operations/observability/mixins/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/operations/observability/mixins/OWNERS)~~ [JanKoehnlein,csweichel,meysholdt]
- ~~[operations/observability/mixins/workspace/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/operations/observability/mixins/workspace/OWNERS)~~ [csweichel]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**